core/src/miner/mem_pool.rs

```diff
@@ -246,12 +245,18 @@ impl MemPool {
             let order = TransactionOrder::for_transaction(&item, client_account.seq);
             let order_with_tag = TransactionOrderWithTag::new(order, QueueTag::New);
 
-            self.by_hash.insert(hash, item);
+            self.by_hash.insert(hash, item.clone());
```

Comment:
You don't need to clone the item if you move this statement to the below of the if statements.

Reply:
Now Items has &[MemPoolItem] type, `item` in for loop has reference type and borrowed from the outer, so I should be reconsidered about the type of the parameter `items`.